src/libraries/System.Text.Json/tests/Serialization/ConstructorTests/ConstructorTests.Cache.cs

```diff
@@ -9,46 +9,47 @@ namespace System.Text.Json.Serialization.Tests
 {
     public abstract partial class ConstructorTests
     {
-        [Fact, OuterLoop]
-        public void MultipleThreadsLooping()
+        [Fact]
+        [OuterLoop]
+        public async Task MultipleThreadsLooping()
         {
             const int Iterations = 100;
 
             for (int i = 0; i < Iterations; i++)
             {
-                MultipleThreads();
+                await MultipleThreads();
             }
         }
 
         [Fact]
-        public void MultipleThreads()
+        public async Task MultipleThreads()
         {
             // Verify the test class has >32 properties since that is a threshold for using the fallback dictionary.
             Assert.True(typeof(ClassWithConstructor_SimpleAndComplexParameters).GetProperties(BindingFlags.Instance | BindingFlags.Public).Length > 32);
 
-            void DeserializeObject(string json, Type type, JsonSerializerOptions options)
+            async Task DeserializeObjectAsync(string json, Type type, JsonSerializerOptions options)
             {
-                var obj = Serializer.Deserialize(json, type, options);
+                var obj = await Serializer.DeserializeAsync(json, type, options);
```

Comment:
Why was this changed to use the Async version of the API? The original intent was to test non-async APIs for thread safety.

Reply:
The original test was calling a synchronous method which was then overridden by two different implementations, one that used the underlying sync API and one that used the underlying async API and then blocked waiting on it.  This change inverts that, so it's an async API that then uses the underlying sync API and wraps it in a task or the underlying async API.  It's done to avoid sync-over-async, which blows up on wasm.

Reply:
Correct. I've tried to capture this in the PR description but please let me know if I could make this clearer.